extensions/fairifier-rdf-extension/dtls/fairifier/DetectFileFormatCommand.java

```diff
@@ -1,145 +1,153 @@
-     * gave problems with other parsers. 
+     * Other formats that can be parsed are Turtle and Ntripples. NOTE: Sesame supports more formats
+     * but the currently used version gave problems with other parsers.
      * 
      * @param req a request object
      * @param res a response object
-     * @see com.google.refine.commands.Command#doPost(javax.servlet.http.HttpServletRequest, javax.servlet.http.HttpServletResponse)
+     * @see com.google.refine.commands.Command#doPost(javax.servlet.http.HttpServletRequest,
+     *      javax.servlet.http.HttpServletResponse)
      */
     @Override
-    public void doPost(HttpServletRequest req, HttpServletResponse res) throws ServletException, IOException {
-        
-        this.parsers = new RDFParser[]{this.rdfParserntriples, this.rdfParserttl};
+    public void doPost(HttpServletRequest req, HttpServletResponse res)
+            throws ServletException, IOException {
+
+        this.parsers = new RDFParser[] {this.rdfParserntriples, this.rdfParserttl,
+                this.rdfParserrdfxml, this.rdfParserJsonld, this.rdfParserNQuads,
+                this.rdfParserJson, this.rdfParserTrix, this.rdfParserTrig};
         FileItemFactory factory = new DiskFileItemFactory();
 
         // Create a new file upload handler
         ServletFileUpload upload = new ServletFileUpload(factory);
         List<FileItem> items = null;
         String baseuri = null;
         String filename = null;
-        InputStream in = null;
+        InputStream[] in = new InputStream[this.parsers.length];
         String format = null;
         String url = null;
-        
-        try{
+
+        try {
             items = upload.parseRequest(req);
-            for(FileItem item:items){
-                if(item.getFieldName().equals("baseuri")){
+            for (FileItem item : items) {
+                if (item.getFieldName().equals("baseuri")) {
                     baseuri = item.getString();
-                }else if (item.getFieldName().equals("file_source")) {
+                } else if (item.getFieldName().equals("file_source")) {
                     url = item.getString();
-                }else if(item.getFieldName().equals("file_upload")){
+                } else if (item.getFieldName().equals("file_upload")) {
                     filename = item.getName();
-                    in = item.getInputStream();
+                    for (int i = 0; i < in.length; i++) {
+                        StringWriter writer = new StringWriter();
+                        in[i] = new BufferedInputStream(item.getInputStream());
+
+                    }
                 }
             }
-        }catch(FileUploadException ex){
+        } catch (FileUploadException ex) {
             respondException(res, ex);
         }
+        StringBuffer response = new StringBuffer();;
 
         if ((url != null) && !url.trim().equals("") && !url.trim().equals("url")) {
```

Comment:
I get that this tries to parse the load and parse the external files. But every single thing in one method is not readable (especially with zero inline comments explaining the flow). This could better be split up into separate methods.